### ISSUE TITLE
For an unknown reason when restarting qTox, Playback volume is written "100" even though it is saved as "30" normally

### ISSUE THREAD

**Issue by TheNain38** [qTox/qTox#2528]
It was saved at 30 and when restarting it is written that:
![playback written 100](https://cloud.githubusercontent.com/assets/13149763/11018220/9d6bc870-85bf-11e5-97f9-3047a3a6e164.PNG)

**Comment by tux3** [comment]
I can't reproduce

**Comment by TheNain38** [comment]
Sometimes I can sometimes not.
Maybe it's when I get an upgraded version of qTox.

**Comment by TheNain38** [comment]
And also, microphone was also set to 61 and not 100, the value is just displayed incorrectly

**Comment by agilob** [comment]
> Sometimes I can sometimes not.

settings are saved every n-seconds, not automatically to prevent overusing your harddrive. could you try changing the settings, wait half a minute and restart?

**Comment by TheNain38** [comment]
@agilob The settings were saved correctly

**Comment by tux3** [comment]
I think I found the bug, it looks like applying a translation always shows 100 no matter what the real number is.

**Comment by TheNain38** [comment]
@tux3 It isn't fixed, but I found the bug. If I use the keyboard to change the value of the sliders then, their values aren't saved, but if I use my mouse and drag it, it is saved. It wasn't translations that were causing the bug.

**Comment by tux3** [comment]
@TheNain38 I can't reproduce that, the keyboard keys work fine for me here.

**Comment by TheNain38** [comment]
If I use the left or the right arrows to change the value
https://github.com/tux3/qTox/blob/9d527e7edd5af92826b6058cbbcf94597ac64cca/src/widget/form/settings/avform.cpp#L67
https://github.com/tux3/qTox/blob/9d527e7edd5af92826b6058cbbcf94597ac64cca/src/widget/form/settings/avform.cpp#L68

**Comment by tux3** [comment]
@TheNain38 So it works when you drag it with your mouse, but not with the left/right arrow keys?
For me all of that works, if it doesn't, that's very strange.

**Comment by TheNain38** [comment]
http://doc.qt.io/qt-5/qabstractslider.html#sliderReleased
It is written:

```
This signal is emitted when the user releases the slider with the mouse, or programmatically when setSliderDown(false) is called.
```

**Comment by tux3** [comment]
Yes, but the text is updated on valueChanged with autoconnect (`AVForm::on_playbackSlider_valueChanged`).
http://doc.qt.io/qt-5/qabstractslider.html#valueChanged

> This signal is emitted when the slider value has changed, with the new slider value as argument.

**Comment by TheNain38** [comment]
Yes, the display of the value works correctly, but if you restart qTox after changing the sliders only using left and right arrows, then on restart thay are the same as before you change them

**Comment by tux3** [comment]
Indeed, let's fix that then.

**Comment by TheNain38** [comment]
And you can revert the last commit it isn't needed

**Comment by tux3** [comment]
It is needed if you change translations, that was a real bug.

**Comment by TheNain38** [comment]
Oh, yes, didn't see that

**Comment by tux3** [comment]
This should be fixed by b6d752fd4b21ad4b22523fe6cc221649b8055305, can you confirm?

**Comment by TheNain38** [comment]
When, jenkins finish building it, I could say. But normally it should be fixed, I see no reason it wouldn't.

**Comment by TheNain38** [comment]
@tux3 You should also change the method's names:
- https://github.com/tux3/qTox/blob/master/src/widget/form/settings/avform.cpp#L67
- https://github.com/tux3/qTox/blob/master/src/widget/form/settings/avform.cpp#L68
- https://github.com/tux3/qTox/blob/master/src/widget/form/settings/avform.cpp#L359
- https://github.com/tux3/qTox/blob/master/src/widget/form/settings/avform.cpp#L364

**Comment by TheNain38** [comment]
Or even better, this: https://github.com/tux3/qTox/blob/master/src/widget/form/settings/avform.cpp#L347
is called when the value change, you could place this line: https://github.com/tux3/qTox/blob/master/src/widget/form/settings/avform.cpp#L361
directly in

**Comment by tux3** [comment]
I merged the 4 slots into 2

**Comment by tux3** [comment]
Fuck. Don't pull that commit.

**Comment by TheNain38** [comment]
This is fixed in b6d752fd4b21ad4b22523fe6cc221649b8055305, thanks!
